samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentEntityProducer.java

```diff
@@ -481,9 +503,22 @@ public void updateEntityReferences(String toContext, Map<String, String> transve
 							}
 						}
 
+						boolean itemFeedbacksChanged = false;
+						if ( item.getItemFeedbackSet() != null && !item.getItemFeedbackSet().isEmpty() ) {
+							for (Iterator<ItemFeedbackIfc> j = item.getItemFeedbackSet().iterator(); j.hasNext(); ) {
```

Comment:
Could we do `for (ItemFeedbackIfc feedback : itemFeedbacks)`? We also don't need to explicitly check for the collection to be empty.

Reply:
Ha ha - I looked at that code and had the same thought - I will fix :)